### PR TITLE
dist/tools/jlink: add -nogui to version test, fix for OSX

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

If multiple debuggers are present when testing for JLinkExe version
a GUI will open to select the debugger to attach, so add -nogui
when testing for version as well.

### Testing procedure

Attempt flashing with multiple jlink devices connected a gui will open. With this PR it does not.

### CODE REVIEW

File: dist/tools/jlink/jlink.sh [hunk 1]
```diff
@@ -157,7 +157,9 @@ test_term() {
 
 test_version() {
     JLINK_MINIMUM_VERSION="6.74"
-    JLINK_VERSION=$(echo q | "$JLINK" 2> /dev/null | grep "^DLL version*" | grep -oP "\d+\.\d+")
+    # Adding '-nogui 1' will simply return' Unknown command line option -nogui'
```
Comment (aabadie) [inline dist/tools/jlink/jlink.sh:160]:
```suggestion
    # Adding '-nogui 1' will simply return 'Unknown command line option -nogui'
```

**Comment by fjmolinas** [comment]
@aabadie @rdagher pointed out that `-P` did not work on OSX, so I replaced with `-E` which is supported in OSX and BSD as well:

- https://ss64.com/osx/grep.html
- https://www.freebsd.org/cgi/man.cgi?query=grep&sektion=&n=1

**Comment by rdagher** [comment]
Tested on macos with success:
```shell
$ grep --version
grep (BSD grep) 2.5.1-FreeBSD

$ JLinkExe
SEGGER J-Link Commander V6.86g (Compiled Nov  6 2020 18:12:16)
DLL version V6.86g, compiled Nov  6 2020 18:12:00

Connecting to J-Link via USB...O.K.
Firmware: J-Link OB-STM32F072-128KB-CortexM compiled Jan 21 2020 17:32:07
Hardware version: V1.00
S/N: 760103336
VTref=3.300V


Type "connect" to establish a target connection, '?' for help
J-Link>q

$ echo q | JLinkExe -nogui 1 2> /dev/null | grep "^DLL version*" | grep -oE '[0-9]+\.[0-9]+'
6.86
```

**Review by aabadie — APPROVED** [review]
Tested with 2 JLink plugged at the same time. I confirm the issue reported and that this PR is fixing it.
I also downgraded JLink to 6.70 and confirm that the version is correctly parsed even if -nogui is not supported.

ACK

**Pull request merged**